src/mock_parsec/tests.rs

```diff
@@ -204,6 +204,77 @@ fn consensus_mode_single() {
     assert_eq!(alice_blocks, bob_blocks);
 }
 
+#[test]
+fn reevaluate_previously_insufficient_votes() {
+    init_mock();
+
+    // Create 5 nodes (A, B, C, D, E). 3 of them vote to remove D, which is not enough to reach
+    // consensus. Then all vote to remove E which gets consensused and E is removed. The previous
+    // votes to remove D are now enough to reach consensus and so D should be removed too.
+
+    let genesis_group: BTreeSet<_> = (0..5).map(|n| PeerId(n)).collect();
+    let mut nodes: Vec<_> = genesis_group
```

Comment:
Maybe worth making that a well named helper function that take a number and a ConsensusMode and return nodes.
And use it for the other 2 tests as well?